imports/plugins/core/graphql/server/schemas/cart.graphql

```diff
@@ -0,0 +1,364 @@
+  priceWhenAdded: Float!
+
+  "Date last updated"
+  updatedAt: DateTime!
+}
+
+input ProductConfigurationInput {
+  "The Product ID"
+  productId: ID!
+
+  "The ProductVariant ID, if there is one"
+  productVariantId: ID
+
+  "An ordered list of product options/attributes that were selected"
+  options: [ProductOptionInput]
+
+  "A title for use in cart/orders that conveys the selected product's title + chosen options"
+  title: String!
+}
+
+"Chosen product option information"
+input ProductOptionInput {
+  "Option or attribute label - e.g. color"
+  label: String
+
+  "Option or attribute value - e.g. blue"
+  value: String
+
+  "Order of this option in the list of options, ascending."
+  position: Int
+}
+
+
+input AddCartItemInput {
+  cartId: ID!,
+  "Array of items to be added to the cart"
+  items: [CartItemInput]!
+}
+
+input RemoveCartItemsInput {
+  cartId: ID!,
+  "Array of items to remove from the cart."
+  cartItemIds: [ID]!
+}
+
+input UpdateCartItemInput {
+  cartItemId: ID!
+  "New absolute value for specified cart item's quantity. Not an incremental value."
+  quantity: Int!
+}
+
+input UpdateCartItemsQuantityInput {
+  cartId: ID!
+  "Array of cart item quantities to update."
+  items: [UpdateCartItemInput]!
+}
+
+extend type Mutation {
+  # Enforce that at least one item is provided so we don't create empty carts.
+  #
+  # IMPORTANT: CartID must be treated as (or more) securely as a session ID. We
+  # need to prevent brute-force cart id collisions.
+  #
+  #   See: https://www.owasp.org/index.php/Session_Management_Cheat_Sheet
+  #
+  #   * ID value must provide at least 64 bits of entropy (if a good PRNG is
+  #     used, this value is estimated to be half the length of the session ID).
+  #   * If a session ID with an entropy of 64 bits is used, it will take an
+  #     attacker at least 292 years to successfully guess a valid session ID,
+  #     assuming the attacker can try 10,000 guesses per second with 100,000
+  #     valid simultaneous sessions available in the web application [2].
+  #
+  "Create a new cart."
+  createCart(input: CreateCartInput!): Cart!
+
+  # TODO: Consider returning the connection for the following mutations to
+  # resolve questions of which type should be returned.
+
+  # Add an item to cart.
+  # If the item already already exists in the cart, then increment quantity.
+  # TODO: Consider returning connection?
+  "Add item(s) to a cart."
+  addCartItems(input: AddCartItemInput!): Cart! # CartItemConnection
+
+  # Once added to cart, items should be referenced by their cart item ID.
+  # Enforce that at least one item is in the list.
+  "Remove item(s) from a cart."
+  removeCartItems(input: RemoveCartItemsInput!): Cart! # CartItemConnection???
+
+  # If cart item input is set to 0, then remove the item.
+  "Update cart item(s) quantity. Use absolute quantity."
+  updateCartItemsQuantity(input: UpdateCartItemsQuantityInput!): Cart! # CartItemConnection
```

Comment:
The response needs to be `UpdateCartItemsQuantityPayload` type, with definition:

```graphql
type UpdateCartItemsQuantityPayload {
  "The modified cart"
  cart: Cart!

  "The same string you sent with the mutation params, for matching mutation calls with their responses"
  clientMutationId: String
}
```